R/zchunk_L202.Ccoef.R

```diff
@@ -30,43 +32,63 @@ module_energy_L202.Ccoef_DISABLED <- function(command, ...) {
     # Load required inputs
     GCAM_region_names <- get_data(all_data, "common/GCAM_region_names")
     A_PrimaryFuelCCoef <- get_data(all_data, "emissions/A_PrimaryFuelCCoef")
-    fuel_to_Ccoef <- get_data(all_data, "emissions/fuel_to_Ccoef")
+    fuel_to_Ccoef <- get_data(all_data, "emissions/mappings/fuel_to_Ccoef")
     L102.Ccoef_kgCGJ_F_Yh <- get_data(all_data, "L102.Ccoef_kgCGJ_F_Yh")
     L102.Ccoef_kgCGJ_R_F_Yh <- get_data(all_data, "L102.Ccoef_kgCGJ_R_F_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Carbon contents of fuels in all regions
+    A_PrimaryFuelCCoef %>%
+      write_to_all_regions(c(LEVEL2_DATA_NAMES[["CarbonCoef"]]), GCAM_region_names = GCAM_region_names, has_traded = TRUE) -> L202.CarbonCoef
+
+
+    # For regions with no agricultural and land use sector (Taiwan),
+    # need to remove the passthrough supplysectors for first-gen biofuels
+    ag_en <- c( "regional corn for ethanol", "regional sugar for ethanol", "regional biomassOil" )
+    L202.CarbonCoef %>%
+      filter(!region %in% aglu.NO_AGLU_REGIONS | !PrimaryFuelCO2Coef.name %in% ag_en) -> L202.CarbonCoef
+
+    # if emissions.USE_GCAM3_CCOEFS = 1, using exogenous C coefs from GCAM 3.0
+    if(emissions.USE_GCAM3_CCOEFS == 0) {
```

Comment:
@kvcalvin @d3y419 Should the data system _ever_ have to use GCAM3 coefficients, going forward? Can this code just be deleted?

Reply:
@pkyle maybe has thoughts? To my knowledge (based on my and Marshall's GCAM-core) we are currently using the GCAM3 coefficients. This has some implications for the OTAQ project if this changes.

Reply:
This looks fine to me. Unlike the other instances of "GCAM3" in the data system, these values are still in the core model. For accuracy they could be renamed "MiniCAM", because that's really where they come from.
We should update that at some point, in order to calibrate CO2 emissions volumes in the base years. I.e., GCAM's CO2 emissions in 1990, 2005, and 2010 are different from all inventories, globally, by region, and by fuel. But that switch might make its own issues; the coefs we have always used are high for coal and low for gas.